src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1052,11 +1055,197 @@ private void AddValueInternal(DateTime time, Span<double> values, Span<byte> tag
+                                    {
+                                        splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        continue;
+                                    }
+
+                                    var compareResult = CompareByTags(localTag, tagSlice);
+
+                                    if (compareResult == CompareResult.Local)
+                                    {
+                                        splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        continue;
+                                    }
+                                    if (compareResult == CompareResult.Equal)
+                                    {
+                                        if (FromReplication == false || localValues.SequenceCompareTo(newValues) < 0)
+                                            splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                        else
+                                            splitSegment.Append(_context.Allocator, 0, localValues, localTag.AsSpan(), localStatus);
+                                        
+                                        isAppend = true;
+                                        continue;
+                                    }
+
+                                    splitSegment.Append(_context.Allocator, 0, newValues, tagSlice, status);
+                                    isAppend = true;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            private static CompareResult CompareByTags(TimeSeriesValuesSegment.TagPointer localTag, Span<byte> tagSlice)
+            {
+                if ((localTag.Pointer == null || localTag.Size == 0) && (tagSlice.IsEmpty)) // both tags are empty
+                    return CompareResult.Equal;
+
+                if (localTag.Pointer == null || localTag.Size == 0) // local tag is empty and remote not
+                    return CompareResult.Remote;
+
+                if ((tagSlice.IsEmpty)) // remote tag is empty and local not
+                    return CompareResult.Local;
+
+                fixed (byte* pTagSlice = &tagSlice.GetPinnableReference())
+                {
+                    var compare = LazyStringValue.CompareToOrdinalIgnoreCase(localTag.Pointer + 1, localTag.Size, pTagSlice + 1, tagSlice.Length);
+                  
+                    return compare switch
+                    {
+                        0 => CompareResult.Equal,
+                        > 0 => CompareResult.Remote,
+                        _ => CompareResult.Local
+                    };
+                }
+            }
+
+            private bool TryMergeEntries(ref TimeSeriesValuesSegment timeSeriesSegment)
```

Comment:
why do we have this and `TryMergeIncrementResults`? should it be the same?

Reply:
no, because here we want to merge all entries into one regardless of the tags.